File: m/rls/stmem.r
file	stmem - string member test
include	rid:rider
include	rid:stdef

code	st_mem - string membership test

  func	st_mem
	cha : int			; the non-zero character
	str : * char~			; the set (or <> for empty set)
	()  : int			; 1=>found, 0=>missing
  is	tmp : char~			;
	fail if str eq <>		; null set
	while (tmp = *str++) ne		; got more
	   fine if <char>tmp eq <char>cha; got a match
	end				;
	fail				;
  end


File: m/rls/rlb/stmem.c
/* file -  stmem - string member test */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_mem - string membership test */
int st_mem(
register int cha ,
char *str )
{ char tmp ;
  if( str == NULL)return 0;
  while ( (tmp = *str++) != 0) {
    if( (BYTE )tmp == (BYTE )cha)return 1;
  } 
  return 0;
} 
